src/measure_theory/measure/haar_lebesgue.lean

```diff
@@ -369,6 +370,55 @@ calc μ (affine_map.homothety x r '' s) = μ ((λ y, y + x) '' (r • ((λ y, y
 ... = ennreal.of_real (abs (r ^ (finrank ℝ E))) * μ s :
   by simp only [image_add_right, measure_preimage_add_right, add_haar_smul]
 
+lemma add_haar_univ [nontrivial E] : μ (univ : set E) = ∞ :=
+begin
+  have A : 0 < μ univ, from is_open_univ.measure_pos μ univ_nonempty,
+  have : tendsto (λ r, ennreal.of_real (abs (r ^ (finrank ℝ E))) * μ univ) at_top (𝓝 (∞ * μ univ)),
+  { simp only [mul_comm _ (μ univ)],
+    refine ennreal.tendsto.const_mul _ (or.inl ennreal.top_ne_zero),
+    apply ennreal.tendsto_of_real_at_top.comp (tendsto_abs_at_top_at_top.comp _),
+    exact tendsto_pow_at_top (ne_of_gt finrank_pos) },
+  simp only [ennreal.top_mul, A.ne', if_false] at this,
+  have : ∞ ≤ μ univ,
+  { apply le_of_tendsto' this (λ r, _),
+    rw ← add_haar_smul,
+    exact measure_mono (subset_univ _) },
+  exact eq_top_iff.2 this,
+end
+
+lemma integral_comp_smul (f : E → F) (R : ℝ) :
```

Comment:
Please add a comment mentioning that this formula is true without integrability assumptions thanks to `0` being used as the "default" value.